addons/binding/org.openhab.binding.wizlighting/src/main/java/org/openhab/binding/wizlighting/internal/discovery/WizLightingDiscoveryService.java

```diff
@@ -0,0 +1,277 @@
+
+        logger.debug("Result {}", urlBuilder.toString());
+        Request request = httpClient.newRequest(urlBuilder.toString()).header("Authorization",
+                "Basic Ml81ZHE1cHFzNnI0b3dvNDAwb284MGNvd2d3azA0Y2drb2t3azB3bzgwa3drb2M4d2NjczpNV0ppWTJFeU1Ea3dNR0V3TVdJM1pqTTFOREppWm1RMU1qTm1NV05qTVRVNE9UZzFOag==")
+                .header("Content-Type", "application/x-www-form-urlencoded").method(HttpMethod.POST)
+                .content(new StringContentProvider("grant_type=urn:anonymous&scope=wiz_ios"));
+        logger.debug("Request is {}", request.toString());
+        ContentResponse response = request.send();
+        logger.debug("Oauth Result {}", response.getContentAsString());
+        TokenDTO data = gson.fromJson(response.getContentAsString(), TokenDTO.class);
+        if (data != null) {
+            accessToken = data.access_token;
+        }
+        return accessToken;
+    }
+
+    protected boolean authorizeInviteToken(String accessToken)
+            throws InterruptedException, TimeoutException, ExecutionException {
+        AuthenticationStore a = httpClient.getAuthenticationStore();
+        a.clearAuthentications();
+        StringBuilder urlBuilder = new StringBuilder(
+                WizLightingBindingConstants.DISCOVERY_API_INVITE_TOKEN_URL + this.inviteToken);
+        logger.debug("Result {}", urlBuilder.toString());
+        Request request = httpClient.newRequest(urlBuilder.toString()).header("Authorization", "Bearer " + accessToken)
+                .timeout(10, TimeUnit.SECONDS);
+        ContentResponse response = request.send();
+        String result = response.getContentAsString();
+        logger.debug("Invite Result {}", result);
+        return result.contains("\"success\":true");
+    }
+
+    protected int retriveHomeId(String accessToken) throws InterruptedException, TimeoutException, ExecutionException {
+        int homeId = -1;
+        AuthenticationStore a = httpClient.getAuthenticationStore();
+        a.clearAuthentications();
+        StringBuilder urlBuilder = new StringBuilder(WizLightingBindingConstants.DISCOVERY_API_USER_DETAILS_URL);
+        logger.debug("Result {}", urlBuilder.toString());
+        Request request = httpClient.newRequest(urlBuilder.toString()).header("Authorization", "Bearer " + accessToken)
+                .timeout(10, TimeUnit.SECONDS);
+        ContentResponse response = request.send();
+        String result = response.getContentAsString();
+        logger.debug("User Details Result {}", result);
+        UserDTO data = gson.fromJson(response.getContentAsString(), UserDTO.class);
+        if (data != null) {
+            homeId = data.getHomeId();
+        }
+
+        return homeId;
+    }
+
+    protected HomeDTO.Light[] retriveHomeLights(String accessToken, int homeId)
+            throws InterruptedException, TimeoutException, ExecutionException {
+        AuthenticationStore a = httpClient.getAuthenticationStore();
```

Comment:
The next 8 lines are the same as in `retrieveHomeId`, please extract them into a method and provide the URL as a parameter and return the `ContentResponse`.